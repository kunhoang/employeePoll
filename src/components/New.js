import "../assest/new.css";
import { connect } from "react-redux";
import { useNavigate } from "react-router-dom";
import { useState } from "react";
import { handleAddQuestion } from "../actions/questions";
import {_getQuestions} from '../util/_DATA.js'
const New = ({ dispatch }) => {
  const [firstOption, setFirstOption] = useState("");
  const [secondOption, setSecondOption] = useState("");

  const handleFirstOptionChange = (e) => {
    const value = e.target.value;
    setFirstOption(value);
  };

  const handleSecondOptionChange = (e) => {
    const value = e.target.value;
    setSecondOption(value);
  };

  const navigate = useNavigate();
  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(handleAddQuestion(firstOption, secondOption));
    navigate("/")
};
  return (
    <div className="new">
      <div className="new-title">Would You Rather</div>
      <div className="description">
        <p>Create your own poll</p>
      </div>
      <div className="first-option">
        <span className="first-option-text">First Option</span>
        <input
          type="text"
          className="text-first-option"
          value={firstOption}
          onChange={handleFirstOptionChange}
        />
      </div>
      <div className="second-option">
        <span className="second-option-text">Second Option</span>
        <input
          type="text"
          className="text-second-option"
          value={secondOption}
          onChange={handleSecondOptionChange}
        />
      </div>
      <button className="btn-submit" onClick={handleSubmit}>Submit</button>
    </div>
  );
};

export default connect()(New);
