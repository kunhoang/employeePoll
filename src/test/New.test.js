import React from 'react';
import {render} from '@testing-library/react';
import New from '../components/New';
import {Provider} from "react-redux";
import {store} from "../store";
import {BrowserRouter} from "react-router-dom";

describe("Home", () => {
    it("Load component New", () => {
        const component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <New/>
                </BrowserRouter>
            </Provider>
        );
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    });
});