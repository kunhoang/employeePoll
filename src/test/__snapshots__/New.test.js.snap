// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Load component Home 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="main-body"
          data-testid="home"
        >
          <div
            class="main"
            data-testid="login"
          >
            <div
              class="title"
            >
              Employee Polls
            </div>
            <div
              class="logo"
            >
              <img
                class="img"
                src="image.jpg"
              />
            </div>
            <div
              class="form"
            >
              <div
                class="form-title"
              >
                Login
              </div>
              <div
                class="form-user"
              >
                <span
                  class="form-user-text"
                >
                  User
                </span>
                <input
                  class="text-user"
                  data-testid="username"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-password"
              >
                <span
                  class="form-password-text"
                >
                  Password
                </span>
                <input
                  class="text-password"
                  data-testid="password"
                  type="password"
                  value=""
                />
              </div>
              <button
                class="btn-submit"
                data-testid="submit"
              >
                Submit
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="main-body"
        data-testid="home"
      >
        <div
          class="main"
          data-testid="login"
        >
          <div
            class="title"
          >
            Employee Polls
          </div>
          <div
            class="logo"
          >
            <img
              class="img"
              src="image.jpg"
            />
          </div>
          <div
            class="form"
          >
            <div
              class="form-title"
            >
              Login
            </div>
            <div
              class="form-user"
            >
              <span
                class="form-user-text"
              >
                User
              </span>
              <input
                class="text-user"
                data-testid="username"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-password"
            >
              <span
                class="form-password-text"
              >
                Password
              </span>
              <input
                class="text-password"
                data-testid="password"
                type="password"
                value=""
              />
            </div>
            <button
              class="btn-submit"
              data-testid="submit"
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Home Load component New 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="new"
      >
        <div
          class="new-title"
        >
          Would You Rather
        </div>
        <div
          class="description"
        >
          <p>
            Create your own poll
          </p>
        </div>
        <div
          class="first-option"
        >
          <span
            class="first-option-text"
          >
            First Option
          </span>
          <input
            class="text-first-option"
            type="text"
            value=""
          />
        </div>
        <div
          class="second-option"
        >
          <span
            class="second-option-text"
          >
            Second Option
          </span>
          <input
            class="text-second-option"
            type="text"
            value=""
          />
        </div>
        <button
          class="btn-submit"
        >
          Submit
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="new"
    >
      <div
        class="new-title"
      >
        Would You Rather
      </div>
      <div
        class="description"
      >
        <p>
          Create your own poll
        </p>
      </div>
      <div
        class="first-option"
      >
        <span
          class="first-option-text"
        >
          First Option
        </span>
        <input
          class="text-first-option"
          type="text"
          value=""
        />
      </div>
      <div
        class="second-option"
      >
        <span
          class="second-option-text"
        >
          Second Option
        </span>
        <input
          class="text-second-option"
          type="text"
          value=""
        />
      </div>
      <button
        class="btn-submit"
      >
        Submit
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
